tensorflow/lite/micro/tools/make/Makefile

```diff
@@ -128,9 +128,11 @@ CCFLAGS  := \
 
 ARFLAGS := -r
 
+ifeq ($(filter armclang,$(ALL_TAGS)),)
```

Comment:
based on https://github.com/tensorflow/tensorflow/issues/43865#issuecomment-705818053, can we have the toolchain be a separate flag instead of being part of `TAGS`?

Copying my suggestion here (and adding in some additional error checking):

```make
TOOLCHAIN :=gcc
ifeq ($(TOOLCHAIN), gcc))
  # default to gcc
  CXX_TOOL := g++
  CC_TOOL := gcc
  AR_TOOL := ar
else ifeq ($(TOOLCHAIN), armclang))
  CXX_TOOL := armclang++?
  CC_TOOL := ?
  AR_TOOL :=?
else
  $(error "Invalid TOOLCHAIN: "$(TOOLCHAIN))
endif
```